src/Razor/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/DefaultDocumentWriter.cs

```diff
@@ -32,7 +32,7 @@ public override RazorCSharpDocument WriteDocument(RazorCodeDocument codeDocument
             }
 
             var context = new DefaultCodeRenderingContext(
-                new CodeWriter(),
+                new CodeWriter(Environment.NewLine, _options.IndentWithTabs, _options.IndentSize),
```

Comment:
Hmmm I wonder if it makes more sense for the code writer to take in the code gen options instead. @pranavkm thoughts now that you own this? 😄

Reply:
I think you're right, it would also make it easier in the future if we were to add other code gen option variables to the constructor.